go/enclave/rollupchain/rollup_chain.go

```diff
@@ -116,6 +116,25 @@ func (rc *RollupChain) insertBlockIntoL1Chain(block *types.Block) error {
 			return fmt.Errorf("block was invalid: %w", err)
 		}
 	}
+
+	// todo: this is minimal L1 tracking/validation, and should be removed when we are using geth's blockchain or lightchain structures for validation
+	l1Latest := rc.storage.FetchHeadBlock()
+
+	// we do a basic sanity check, comparing the received block to the latest known block on the chain
+	if l1Latest == nil {
```

Comment:
is this block here exhaustive?
Can I suggest rewriting it for clarity to highlight all possible cases to make it easy to reason about?

Cases: 
- first l1 block ( no existing head)
- existing head is the parent of the submitted block
- existing head is ahead of the submitted block on the same branch
- existing head is on a fork of the submitted block
- no apparent link between existing head and submitted block ( missing info)
- ??

Reply:
I guess my thinking was that:

a. These distinctions are trivial with a blockchain structure to use for LCA etc. but if we want to punt that work for now then we'd have to walk the tree ourselves (not difficult but it will get thrown away when we switch to chain struct)

b. I don't think the enclave will be behaving differently for these different distinctions. Either it accepts the block as the next block on its chain or it rejects it and says: "this is where I am"

(FWIW the code on the host-side that provides blocks does make these distinctions, it has to figure out what to feed it next based on LCA etc.)

So would you prefer:
1. Walk the tree to distinguish the different cases (clearer logging for the different cases and it might be relevant for the L2 state?)
2. Do the work now to get it working with geth LightChain structure so we can use its LCA etc.
3. Leave it as-is but make it clearer that some of these cases are getting grouped together for now

Reply:
As discussed, I'm going for 1. Distinguishing betweeen the cases and using a `BlockIngestionType` or similar to provide details about the sort of block update we're dealing with.